packages/stores/src/queries/pool-incentives/incentivized-pools.ts

```diff
@@ -119,51 +121,54 @@ export class ObservableQueryIncentivizedPools extends ObservableChainQuery<Incen
   );
 
   /**
-   * Computes external incentive APY for the given duration
+   * Computes the external incentive APR for the given duration
    */
-  readonly computeExternalIncentiveAPYForSpecificDuration = computedFn(
+  readonly computeExternalIncentiveGaugeAPR = computedFn(
     (
       poolId: string,
+      gaugeId: string,
+      denom: string,
       duration: Duration,
       priceStore: IPriceStore,
-      fiatCurrency: FiatCurrency,
-      allowedGauges: ExternalGauge[]
+      fiatCurrency: FiatCurrency
     ): RatePretty => {
-      if (!allowedGauges.length) {
+      const observableGauge = this.queryGauge.get(gaugeId);
+
+      if (
+        duration.asMilliseconds() !==
```

Comment:
I'm realizing that since we take in the gaugeId, there's no real reason to take in the duration as well and perform this check. Can we remove the duration from the params?

Reply:
@jonator Good catch, working on it =D

Reply:
Please make sure the comment/docs is updated as well.

Reply:
I think the main reason why we need duration is - on the pool page - we're iterating all through the `ExternalIncentiveGaugeAllowList[pool.id]`. Would you like me move duration checking there?